src/components/Chart/Lines.layout.js

```diff
@@ -200,43 +204,47 @@ export default props => {
   let endLabelSizes = []
   let yNeedsConnectors = false
 
-  const highlight = props.highlight
-    ? unsafeDatumFn(props.highlight)
-    : () => false
-  const stroke = props.stroke ? unsafeDatumFn(props.stroke) : () => false
   const labelFilter = props.labelFilter
     ? unsafeDatumFn(props.labelFilter)
     : () => true
+
+  const addLabels = (color, colorAccessor, labelFilter, yFormat, y) => ({
+    values: line
+  }) => {
+    const start = line[0]
+    const end = line[line.length - 1]
+    const label = labelFilter(start.datum)
+
+    const isHighlight = props.highlight
+      ? unsafeDatumFn(props.highlight)
+      : () => false
+    const hasStroke = props.stroke ? unsafeDatumFn(props.stroke) : () => false
+
+    return {
+      line,
+      start,
+      end,
+      highlighted: isHighlight(start.datum),
+      stroked: hasStroke(start.datum),
+      lineColor: color(colorAccessor(start)),
+      startValue: label && props.startValue && yFormat(start.value),
+      endValue: label && props.endValue && yFormat(end.value),
+      endLabel: label && props.endLabel && ` ${colorAccessor(end)}`,
+      startY: y(start.value),
+      endY: y(end.value)
+    }
+  }
+
   groupedData = groupedData
-    .map(({ values, key }) => ({
-      key,
-      values: groupBy(values, lineGroup)
-    }))
+    .map(groupByLines(props.color, props.category))
     .map(({ values: lines, key }) => {
-      const linesWithLabels = lines.map(({ values: line }) => {
-        const start = line[0]
-        const end = line[line.length - 1]
-
-        const label = labelFilter(start.datum)
-
-        return {
-          line,
-          start,
-          end,
-          highlighted: highlight(start.datum),
-          stroked: stroke(start.datum),
-          lineColor: color(colorAccessor(start)),
-          startValue: label && startValue && yFormat(start.value),
-          endValue: label && props.endValue && yFormat(end.value),
-          endLabel: label && endLabel && ` ${colorAccessor(end)}`,
-          startY: y(start.value),
-          endY: y(end.value)
-        }
-      })
+      const linesWithLabels = lines.map(
+        addLabels(color, colorAccessor, labelFilter, yFormat, y, props)
+      )
       let labelsMoved = 0
       labelsMoved += calculateAndMoveLabelY(linesWithLabels, 'start')
       labelsMoved += calculateAndMoveLabelY(linesWithLabels, 'end')
-      yNeedsConnectors = yNeedsConnectors || labelsMoved > 1
+      yNeedsConnectors = yNeedsConnectors || labelsMoved >= 1
```

Comment:
this currently leads to connectors also showing up with two lines when labels were just move a tiny bit
<img width="1636" alt="Screenshot 2021-06-15 at 12 22 40" src="https://user-images.githubusercontent.com/410211/122037247-ae8a6b80-cdd4-11eb-8f04-f383f328ebd8.png">
(bottom chart «Mittelstandsinitiative» the second label was moved slightly)

IMO we should only do it when we have more than two lines:
```suggestion
      yNeedsConnectors = yNeedsConnectors || (labelsMoved >= 1 && linesWithLabels.length > 2)
```